theano/gpuarray/opt.py

```diff
@@ -398,6 +409,10 @@ def apply(self, fgraph):
             elif isinstance(new_ops, theano.Variable):
                 outputs = [new_ops]
 
+            for old_output, new_output in zip(node.outputs, outputs):
+                copy_stack_trace(old_output, new_output)
+                new_output.tag.tracefrom = old_output
+
```

Comment:
Whoops, that's debugging code I didn't mean to commit. It allows my to track back where the lack of trace comes from.